### PR TITLE
[SYCL][L0] check for no event leaks

### PR DESCRIPTION
**Pull request by smaslov-intel**
Signed-off-by: Sergey V Maslov <sergey.v.maslov@intel.com>

### CODE REVIEW

File: SYCL/Plugin/level-zero-event-leak.cpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// REQUIRES: level_zero, level_zero_dev_kit
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %level_zero_options %s -o %t.out
+// RUN: env SYCL_DEVICE_FILTER=level_zero ZE_DEBUG=4 %GPU_RUN_PLACEHOLDER %t.out
+//
```
Comment (kbsmith-intel) [inline SYCL/Plugin/level-zero-event-leak.cpp:4]: I think you should have a comment here describing the test, and what it is trying to check for, and why the single CHECK-NOT line for LEAK is sufficient.

File: SYCL/Plugin/level-zero-event-leak.cpp [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+// CHECK-NOT: ---> LEAK
+
+#include <CL/sycl.hpp>
+using namespace cl;
+int main(int argc, char **argv) {
+  sycl::queue Q;
+  const unsigned n_chunk = 129;
```
Comment (kbsmith-intel) [inline SYCL/Plugin/level-zero-event-leak.cpp]: Is 129 a magic number in any way?  I think that also deserves a comment.  Or maybe you could take that as a command line parameter, and run the executable multiple times with different values, and check for leakage with 3 or 4 different values?

**Review by kbsmith-intel — CHANGES_REQUESTED** [review]
Overall, this test makes me wonder if we should also create or use other tests, and add a LEAK check to them.

File: SYCL/Plugin/level-zero-event-leak.cpp (hunk 1 above)
Comment (smaslov-intel) [reply SYCL/Plugin/level-zero-event-leak.cpp:4]: added comment

File: SYCL/Plugin/level-zero-event-leak.cpp (hunk 2 above)
Comment (smaslov-intel) [reply SYCL/Plugin/level-zero-event-leak.cpp]: added comment

**Comment by smaslov-intel** [comment]
> Overall, this test makes me wonder if we should also create or use other tests, and add a LEAK check to them.

We already have testing with ZE_DEBUG=-1, which should check the LEAKs too. I will contact QA to see if it's actually running.

File: SYCL/Plugin/level-zero-event-leak.cpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+//                 zeFenceCreate = 2     \--->                zeFenceDestroy = 0     ---> LEAK = 2
+//                 zeImageCreate = 0     \--->                zeImageDestroy = 0
+//               zeSamplerCreate = 0     \--->              zeSamplerDestroy = 0
+//              zeMemAllocDevice = 0     |
+//                zeMemAllocHost = 0     |
+//              zeMemAllocShared = 0     \--->                     zeMemFree = 0
+//
+// clang-format on
+//
+// NOTE: The 129 value below is to be larger thatn the "128" heuristic in
```
Comment (kbsmith-intel) [inline SYCL/Plugin/level-zero-event-leak.cpp]: If the 128 heuristic in addSharedEvernt is changed to be larger (which it probably should be) then will this test
need to be updated?

Should the value be chosen to be MUCH larger than the current heuristic?

Comment (smaslov-intel) [reply SYCL/Plugin/level-zero-event-leak.cpp]: OK, put 1000.

**Review by kbsmith-intel — APPROVED** [review]
LGTM

**Pull request merged**